src/color.cc

```diff
@@ -276,6 +505,49 @@ rgba_from_rgba(uint8_t r, uint8_t g, uint8_t b, uint8_t a) {
 }
 
 /*
+ * Helper function used in rgba_from_hsla().
+ */
+
+static float hue_to_rgb(float t1, float t2, float hue) {
+  if(hue < 0) hue += 6;
+  if(hue >= 6) hue -= 6;
+
+  if(hue < 1) return (t2 - t1) * hue + t1;
+  else if(hue < 3) return t2;
+  else if(hue < 4) return (t2 - t1) * (4 - hue) + t1;
+  else return t1;
+}
+
+/*
+ * Return rgba from (h,s,l,a).
+ */
+
+static inline int32_t
+rgba_from_hsla(float h_deg, float s, float l, float a) {
+  float h = (6 * h_deg) / 360.0f;
+  float m1,m2;
+  if (l<=0.5)
```

Comment:
whitespace needed on the next 5 lines, don't bunch things up so much like a classic C programmer, we're JavaScripters here!